sdk/metric/meter.go

```diff
@@ -60,6 +63,10 @@ var _ metric.Meter = (*meter)(nil)
 // options. The instrument is used to synchronously record increasing int64
 // measurements during a computational operation.
 func (m *meter) Int64Counter(name string, options ...metric.Int64CounterOption) (metric.Int64Counter, error) {
+	if err := validateInstrumentName(name); err != nil {
+		return nil, err
```

Comment:
A working instrument can be returned with the error here. I think that is the correct behavior.